PxeReaderSDK/Documentation/PxeReaderAPI/man/man3/PxePlayerAnnotation.3
.TH "PxePlayerAnnotation" 3 "Mon Apr 28 2014" "PxeReaderAPI" \" -*- nroff -*-
.ad l
.nh
.SH NAME
PxePlayerAnnotation \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#import <PxePlayerAnnotation\&.h>\fP
.PP
Inherits NSObject\&.
.SS "Properties"

.in +1c
.ti -1c
.RI "NSString * \fBuri\fP"
.br
.ti -1c
.RI "NSString * \fBrange\fP"
.br
.ti -1c
.RI "NSString * \fBcolor\fP"
.br
.ti -1c
.RI "NSString * \fBtext\fP"
.br
.ti -1c
.RI "NSString * \fBnoteText\fP"
.br
.ti -1c
.RI "NSString * \fBcontentId\fP"
.br
.ti -1c
.RI "NSString * \fBannotationDttm\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
A simple class to parse the annotation data into the model 
.PP
Definition at line 14 of file PxePlayerAnnotation\&.h\&.
.SH "Property Documentation"
.PP 
.SS "- (NSString*) annotationDttm\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the annotation definition 
.PP
Definition at line 49 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) color\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the color value of the annotation highlight color 
.PP
Definition at line 29 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) contentId\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the content id 
.PP
Definition at line 44 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) noteText\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the annotation description text 
.PP
Definition at line 39 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) range\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the range of the annotation text 
.PP
Definition at line 24 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) text\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the annotation title text 
.PP
Definition at line 34 of file PxePlayerAnnotation\&.h\&.
.SS "- (NSString*) uri\fC [read]\fP, \fC [write]\fP, \fC [nonatomic]\fP, \fC [strong]\fP"
A NSString variable to hold the uri 
.PP
Definition at line 19 of file PxePlayerAnnotation\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for PxeReaderAPI from the source code\&.


PxeReaderSDK/PxePlayerApi/PxePlayerAnnotation.m
//
//  PxePlayerAnnotation.m
//  PxeReader
//
//  Created by Saro Bear on 07/11/13.
//  Copyright (c) 2013 Pearson. All rights reserved.
//

#import "PxePlayerAnnotation.h"
#import "PXEPlayerMacro.h"
#import "NSString+Extension.h"

@implementation PxePlayerAnnotation

-(id)init
{
    if((self = [super init]))
    {
        self.range = @"";
        self.colorCode = @"";
        self.noteText = @"";
        self.selectedText = @"";
        self.contentId = @"";
        self.annotationDttm = @"";
        self.labels = [NSArray new];
        self.appData = [NSDictionary new];
    }
    return self;
}

- (id) initWithDictionary:(NSDictionary*)annoDict
{
    if((self = [super init]))
    {
        self.annotationDttm = [annoDict objectForKey:@"annotationDttm"];
        DLog(@"self.annotationDttm: %@", self.annotationDttm);
        self.shareable = [[annoDict objectForKey:@"shareable"] boolValue];
        
        NSDictionary *data = [annoDict objectForKey:@"data"];
        
        if(data)
        {
            self.range = [data objectForKey:@"range"];
            self.colorCode = [data objectForKey:@"colorCode"];
            self.noteText = [data objectForKey:@"noteText"];
            self.selectedText = [data objectForKey:@"selectedText"];
            self.labels = [data objectForKey:@"labels"];
        }
        self.contentId = @"";
        self.appData = [NSDictionary new];
    }
    return self;
}

-(void)dealloc
{
    self.range = nil;
    self.colorCode = nil;
    self.noteText = nil;
    self.selectedText = nil;
    self.contentId = nil;
    self.annotationDttm = nil;
    self.labels = nil;
    self.appData = nil;
}

- (NSString *) description
{
    NSMutableString *annotationDesc = [NSMutableString stringWithString:@"PxePlayerAnnotation: \n"];
    [annotationDesc appendFormat:@"   annotationDttm: %@ \n", self.annotationDttm];
    [annotationDesc appendFormat:@"   uri: %@ \n", self.uri];
    [annotationDesc appendFormat:@"   colorCode: %@ \n", self.colorCode];
    [annotationDesc appendFormat:@"   isMathML: %@ \n", self.isMathML?@"YES":@"NO"];
    [annotationDesc appendFormat:@"   labels: %@ \n", self.labels];
    [annotationDesc appendFormat:@"   noteText: %@ \n", self.noteText];
    [annotationDesc appendFormat:@"   selectedText: %@ \n", self.selectedText];
    [annotationDesc appendFormat:@"   range: %@ \n", self.range];
    [annotationDesc appendFormat:@"   contentId: %@ \n", self.contentId];
    
    [annotationDesc appendFormat:@"   shareable: %@ \n", self.shareable?@"YES":@"NO"];
    
    [annotationDesc appendFormat:@"   appData: %@ \n", self.appData];
    
    return annotationDesc;
}

- (NSString *) asJSONString
{
    NSDictionary *jsonDict = [self asDictionaryForJSON];
    
    NSData* jsonData = [NSJSONSerialization dataWithJSONObject:jsonDict options:0 error:nil];
    
    return [[NSString alloc] initWithBytes:[jsonData bytes] length:[jsonData length] encoding:NSUTF8StringEncoding];
}

- (NSDictionary *) asDictionaryForJSON
{
    //    NSDictionary *dataDict = @{
    //                               @"colorCode":self.colorCode,
    //                               @"noteText":self.noteText,
    //                               @"isMathMl":self.isMathML ? @YES : @NO,
    //                               @"range":self.range,
    //                               @"selectedText":self.selectedText,
    //                               @"labels":[self.labels]
    //                               };
    NSMutableDictionary *dataDict = [NSMutableDictionary new];
    if (self.colorCode) {
        [dataDict setObject:self.colorCode forKey:@"colorCode"];
    }
    if (self.noteText) {
        [dataDict setObject:self.noteText forKey:@"noteText"];
    }
    if (self.isMathML) {
        [dataDict setObject:(self.isMathML ? @YES : @NO) forKey:@"expLabel"];
    }
    if (self.range) {
        [dataDict setObject:self.range forKey:@"range"];
    }
    if (self.selectedText) {
        [dataDict setObject:self.selectedText forKey:@"selectedText"];
    }
    if (self.labels) {
        [dataDict setObject:self.labels forKey:@"labels"];
    }
    DLog(@"self.annotationDttm: %@", self.annotationDttm);
    NSDictionary *jsonDict = @{
                               @"annotationDttm":self.annotationDttm,
                               @"shareable":self.shareable? @YES : @NO,
                               @"data":dataDict
                               };
    return jsonDict;
}

- (void) setNoteText:(NSString *)note
{
    if (note)
    {
//        note = [note urlDecodeUsingEncoding];
        note = [note unEscapeCharactersInString];
    }
    _noteText = note;
}

- (void) setDecodedNoteText:(NSString*)note
{
    if (note)
    {
        _noteText = note;
    }
}

//- (void) setColorCode:(NSString *)code
//{
//    if([code rangeOfString:@"#"].location != NSNotFound)
//    {
//        if ([code isEqualToString:@"#FFD231"]) // Yellow
//        {
//            _colorCode = @"0";
//        }
//        else if ([code isEqualToString:@"#FB91CF"]) // Pink
//        {
//            _colorCode = @"1";
//        }
//        else if ([code isEqualToString:@"#54DF48"]) // Green
//        {
//            _colorCode = @"2";
//        }
//        else if ([code isEqualToString:@"#66ffff"]) // Blue
//        {
//            _colorCode = @"2";
//        }
//    }
//    else
//    {
//        _colorCode = code;
//    }
//}

/*
 KEEP UNTIL UPLOADING ANNOTATIONS IS WORKING
 
 @"colorCode":self.colorCode,
 @"noteText":self.noteText,
 @"expLabel":self.isMathML ? @"true" : @"",
 @"shareable":self.shareable?@"true":@"false",
 annotation.colorCode = (long)_shareable.tag>0? @"3" : [NSString stringWithFormat:@"%ld",(long)_currentColor];
 annotation.noteText = noteText;
 annotation.isMathML = _mathML ? _mathML.text : @"";
 annotation.shareable = (long)_shareable.tag>0?@"true":@"false";
 */


@end